src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/api/FilterResourceApiTest.java

```diff
@@ -1,56 +1,230 @@
+        FilterResource filterResourceMetadata = createGenericFilterResourceMetadata(
+            accessorId, resourceId, relationship);
+        Response response = amApiClient.filterResource(filterResourceMetadata)
+            .post(amApiClient.getAccessUrl() + api + version + filterResouce);
+
+        //Then I can get Filter Envelope with wild card & Explicit permissions merged
+        response.then()
+            .assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("resource.data.name", Matchers.equalTo("test"))
+            .body("access.permissions.values()[0].size()", Matchers.equalTo(2))
+            .body("access.permissions.values()[0]", Matchers.containsInAnyOrder(read, update))
+            .log();
+    }
+
+    @Test
+    public void verifyFilterResourceWhenWildCardPermissionRevoked() {
+        //Given When User wants Filter wild card access to resource but wildcard access revoked
+        createExplicitGrantForFilterCase(resourceId, "*", DEFAULT, null, READ);
+        ExplicitAccessMetadata explicitAccessMetadata = ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId("*")
+            .accessorType(DEFAULT)
+            .attribute(attribute)
+            .serviceName(serviceName)
+            .resourceName(resourceName)
+            .resourceType(resourceType)
+            .build();
+        amApiClient.revokeResourceAccess(explicitAccessMetadata);
+
+        //When I call Get access Filter API
+        FilterResource filterResourceMetadata = createGenericFilterResourceMetadata(
+            accessorId, resourceId, relationship);
+        Response response = amApiClient.filterResource(filterResourceMetadata)
+            .post(amApiClient.getAccessUrl() + api + version + filterResouce);
+
+        //Then I can get Filter Envelope should be empty
+        response.then()
+            .assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .contentType(Matchers.isEmptyOrNullString())
+            .log();
+    }
+
+    @Test
+    public void verifyFilterResourceChecksBothResourceIdAndResourceType() {
+
+        //GIVEN there is an explicit access record for a resource
+        createExplicitGrantForFilterCase(resourceId, accessorId, accessorType, relationship, READ);
+        ExplicitAccessGrant.builder()
+            .resourceId(resourceId)
+            .resourceDefinition(ResourceDefinition.builder()
+                .serviceName(serviceName)
+                .resourceName(resourceName)
+                .resourceType(otherResourceType)
+                .lastUpdate(Instant.now())
+                .build())
+            .accessorIds(ImmutableSet.of(accessorId))
+            .accessorType(accessorType)
+            .relationship(relationship)
+            .attributePermissions(ImmutableMap.of(JsonPointer.valueOf(""), ImmutableSet.of(UPDATE)))
+            .lastUpdate(Instant.now())
+            .build();
```

Comment:
is this required to create an ExplicitAccessGrant? does not seem to be used anywhere?